### PR TITLE
[Mbstring ] fix mb_substr return type

### PR DESCRIPTION
**Pull request by orklah**
WIP. Trying to fix issue #182

### CODE REVIEW

**Comment by orklah** [comment]
So, contrary to what I thought when I created #182 , substr can also return false on newer versions of PHP7. The doc only explain that when $start is EQUAL to string length, PHP5 returned false and PHP7 return empty string. When $start is superior, substr continue to return false.

This is not the way mb_substr works. mb_substr always return an empty string when $start is equal or superior to string length

I added two tests to each encoding that happen to call substr internally (ASCII and CP850). 
Every test should fail on PHP5. On PHP7, only the tests with $start longer than the string should fail.

Here is the 3v4l.org link which I used to display my results
https://3v4l.org/QKdXP
(CP850 seems to be missing before PHP5.2.17, but given that this package minimum required version is 5.3.3, I ignored this issue)

I will now begin to modify the mb_substr method to pass the tests

**Comment by orklah** [comment]
The test are passing.

Please tell me if there is something left to do or if there's something I did wrong. This is one of my first PR so I may have made mistakes.

Thanks

File: src/Mbstring/Mbstring.php [hunk 1]
```diff
@@ -546,7 +546,11 @@ public static function mb_substr($s, $start, $length = null, $encoding = null)
     {
         $encoding = self::getEncoding($encoding);
         if ('CP850' === $encoding || 'ASCII' === $encoding) {
-            return substr($s, $start, null === $length ? 2147483647 : $length);
```
Comment (nicolas-grekas) [inline src/Mbstring/Mbstring.php]: `return (string) substr($s, $start, null === $length ? 2147483647 : $length);`

File: tests/Mbstring/MbstringTest.php [hunk 2]
```diff
@@ -257,9 +257,13 @@ public function testSubstr()
         $this->assertSame("\xFF", mb_substr("\x00\xFF", -1, 1, 'ASCII'));
         $this->assertSame("\x00", mb_substr("\x00\xFF", 0, 1, 'ASCII'));
         $this->assertSame("\x00\xFF", mb_substr("\x00\xFF", 0, 2, 'ASCII'));
+        $this->assertSame("", mb_substr("\x00\xFF", 2, 1, 'ASCII'));
+        $this->assertSame("", mb_substr("\x00\xFF", 3, 1, 'ASCII'));
         $this->assertSame("\xFF", mb_substr("\x00\xFF", -1, 1, 'CP850'));
         $this->assertSame("\x00", mb_substr("\x00\xFF", 0, 1, 'CP850'));
         $this->assertSame("\x00\xFF", mb_substr("\x00\xFF", 0, 2, 'CP850'));
+        $this->assertSame("", mb_substr("\x00\xFF", 2, 1, 'CP850'));
+        $this->assertSame("", mb_substr("\x00\xFF", 3, 1, 'CP850'));
```
Comment (nicolas-grekas) [inline tests/Mbstring/MbstringTest.php]: please use single quotes for the empty string

File: src/Mbstring/Mbstring.php (hunk 1 above)
Comment (orklah) [reply src/Mbstring/Mbstring.php]: Done :)

File: tests/Mbstring/MbstringTest.php (hunk 2 above)
Comment (orklah) [reply tests/Mbstring/MbstringTest.php]: Yes, my bad!

**Comment by fabpot** [comment]
Thank you @orklah.

**Pull request merged**